File: library/src/main/java/com/anthonycr/base47/Preconditions.java
package com.anthonycr.base47;


import org.jetbrains.annotations.Nullable;

/**
 * Preconditions checking.
 * <p>
 * Created by restainoa on 12/21/16.
 */
final class Preconditions {

    private Preconditions() {
        throw new UnsupportedOperationException("Class is not instantiable");
    }

    static void checkNotNull(@Nullable Object object) {
        if (object == null) {
            throw new NullPointerException("Object must not be null");
        }
    }

}

File: library/src/main/java/com/anthonycr/base47/Preconditions.kt
package com.anthonycr.base47

/**
 * Preconditions checking.
 *
 *
 * Created by restainoa on 12/21/16.
 */
internal class Preconditions private constructor() {

    companion object {

        fun checkNotNull(`object`: Any?) {
            if (`object` == null) {
                throw NullPointerException("Object must not be null")
            }
        }
    }

}
